test/models/student_test.rb

```diff
@@ -15,6 +15,34 @@ class StudentTest < ActiveSupport::TestCase
     assert_equal classroom, student.classroom
   end
 
+  test "has many classrooms through enrollments" do
+    student = create(:student)
+    initial_classroom_count = student.classrooms.count # Factory creates 1 default classroom
+    classroom1 = create(:classroom)
+    classroom2 = create(:classroom)
+
+    student.enrollments.create!(classroom: classroom1)
+    student.enrollments.create!(classroom: classroom2)
+
+    assert_equal initial_classroom_count + 2, student.classrooms.count
+    assert_includes student.classrooms, classroom1
+    assert_includes student.classrooms, classroom2
+  end
+
+  test "can have archived classroom in history" do
+    student = create(:student)
+    initial_count = student.enrollments.count # Factory creates 1 default join record
```

Comment:
That's the reason I would avoid adding after create to the factory. 

It's much easier if you need to add it, and you don't need to remember it. 

What do you think?